app/connectors/CreateIncomeSourceConnector.scala

```diff
@@ -17,59 +17,60 @@
 package connectors
 
 import config.FrontendAppConfig
-import models.addIncomeSource._
+import models.createIncomeSource._
 import play.api.Logger
-import play.api.http.Status.{INTERNAL_SERVER_ERROR, OK}
-import play.api.libs.json.Json
-import uk.gov.hmrc.http.{HeaderCarrier, HttpClient}
+import play.api.http.Status.OK
+import play.api.libs.json.{JsError, Json}
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
 
-import javax.inject.Inject
+import javax.inject.{Inject, Singleton}
 import scala.concurrent.{ExecutionContext, Future}
 
+@Singleton
 class CreateIncomeSourceConnector @Inject()(val http: HttpClient,
                                             val appConfig: FrontendAppConfig
                                            )(implicit val ec: ExecutionContext) {
 
+  def createBusinessIncomeSourcesUrl(mtdItid: String): String =
+    s"${appConfig.itvcProtectedService}/income-tax-view-change/create-income-source/business/$mtdItid"
 
-  def addBusinessDetailsUrl(mtdItid: String): String = s"${appConfig.itvcProtectedService}/income-tax-view-change/create-income-source/business/$mtdItid"
+  def createBusiness(mtdItid: String, request: CreateBusinessIncomeSourceRequest)
+                        (implicit headerCarrier: HeaderCarrier): Future[Either[CreateIncomeSourceErrorResponse, List[CreateIncomeSourceResponse]]] = {
+    val url = createBusinessIncomeSourcesUrl(mtdItid)
+    val jsonRequest = Json.toJson(request)
 
-  def createBusiness(mtdItid: String, createBusinessIncomeSourcesRequest: CreateBusinessIncomeSourceRequest)
-                    (implicit headerCarrier: HeaderCarrier): Future[Either[CreateBusinessErrorResponse, List[AddIncomeSourceResponse]]] = {
-    val bodyAsJson = Json.toJson(createBusinessIncomeSourcesRequest)
-    val url = addBusinessDetailsUrl(mtdItid)
-    http.POST(url, bodyAsJson).map {
-      case response if response.status == OK =>
-        response.json.validate[List[AddIncomeSourceResponse]].fold(
-          _ => {
-            Logger("application").error(s"[CreateIncomeSourceConnector][createBusiness] - Json validation error parsing repayment response, error ${response.body}")
-            Left(CreateBusinessErrorResponse(response.status, s"Not valid json: ${response.body}"))
-          },
-          valid =>
-            Right(valid)
-        )
-      case response =>
-        Logger("application").error(s"[CreateIncomeSourceConnector][createBusiness] - Response status: ${response.status}, body: ${response.body}")
-        Left(CreateBusinessErrorResponse(response.status, s"Error creating incomeSource: ${response.json}"))
-    }
+    http.POST(url, jsonRequest).flatMap(handleResponse)
+  }
+
+  def createForeignProperty(mtdItid: String, request: CreateForeignPropertyIncomeSourceRequest)
```

Comment:
this is fine but perhaps something like this would be better:
`def createBusiness(mtdItid: String, a: Option[CreateBusinessIncomeSourceRequest], b: Option[FPRequest], c: Option[UKPRequest]) = {
val req = Array(a, b, c).flatten.head
vall json = Json.toJson(req)
}`